### DISCUSSION TITLE
[2-1] 오류가 없는데 정상작동이 안되는 경우

### DISCUSSION THREAD

**Discussion by moonuckjoo** [bbang-jun/2023-OOP-Mentoring#10] [Q&A]
#define _CRT_SECURE_NO_WARNINGS
#include <iostream>
#include <cstring>

using namespace std;
class Student_info {
private:
	char name[10];
	int age = 0;
	char university[100];
	char major[100];

public: // 값을 입력받아서 private에 저장
	Student_info(char* name, int age, char* univ, char* major)
	{
		strcpy(this->name, name);
		strcpy(this->university, univ);
		strcpy(this->major, major);
		this->age = age;
	};
	bool find(char* name)
	{
		if (strcmp(this->name, name) == 0)
			return 0;
		else
			return 1;
	};
	void print()  //private의 저장된 것 출력
	{
		cout << "----------print----------";
		cout << "Name : " << this->name << endl;
		cout << "Age : " << this->age << endl;
		cout << "University : " << this->university << endl;
		cout << "Major : " << this->major << endl;
		cout << "-------------------------" << endl;
	};
	void change(char* name, int age, char* univ, char* major)
	{
		if (strcmp(this->name, name) == 0)
		{
			strcpy(this->name, name);
			strcpy(this->university, univ);
			strcpy(this->major, major);
			this->age = age;
		}
	};
	~Student_info() {
	};
	};
	int main (void)
	{
		Student_info* Student[10] = {0};
		char input[10] ={0}; // 커맨드 저장
			// 클래스 메소드에 전달할 정보 저장
		char mainname[10] = { 0 };
		int mainage = 0;
		char mainuniv[100] = { 0 };
		char mainmajor[100] = { 0 };
		char findname[10] = { 0 };
		int i = 0;
		
		while (1) // 무한 반복
		{
			cout << "Please Enter Command(Insert, find, change, print, exit) : ";
			cin >> input; // 커맨드를 입력받음
			if (strcmp(input,"insert")==0) // insert인 경우
			{	
				cin >> mainname >> mainage >> mainuniv >> mainmajor;
				Student[i] = new Student_info(mainname, mainage, mainuniv, mainmajor); 
				i++; // 다음 사람거
			}
			else if (strcmp(input, "find") == 0)
			{
				cin >> findname;
				for (int j = 0; j < 10; j++) {
					if (Student[i]->find(findname) == 0)
						Student[i]->print();
				}
			}
			else if (strcmp(input, "change"))
			{
				for (int l = 0; l < 10; l++)
				{
					Student[l]->change(findname, mainage, mainuniv, mainmajor);
				}
			}
			else if (strcmp(input, "print") == 0)
			{
				for(int k=0;k<10;k++)
					Student[k]->print();
			}
			else if (i > 9 || strcmp(input, "exit"))
				break;

			else
			{
				cout << "Please Enter Command(insert, find, change, print, exit) : ";
				cin >> input; // 커맨드를 입력받음
			}
		}
		delete Student;
		return 0;
	}
제 코드입니다. 아직 손을 덜 보긴 했는데 일단 오류가 발생하진 않고 insert까지는 잘 돌아갑니다.
그러나 print를 입력하는 경우 메인 함수의 else if 문에서 change인 경우에 참이라고 작동이 되어버립니다.. 어떻게 된 일일까요 ㅠㅠㅠㅠ 문제가 없다고 나오기도 하고 단계별로 디버깅 해도 별다른 말이 안 나와서 어떤게 문제인지 모르겠습니다

**Comment by bbang-jun** [comment]
올바른 문법을 작성하시지 않으셔서 런타임 시에 이상한 로직이 작동하는 것 같습니다. 확인 부탁드립니다.
